File: banking-frontend/src/app/core/services/account.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { environment } from '../../../environments/environment';
import { Account, CreateAccountRequest, UpdateAccountRequest } from '../models/account.model';

@Injectable({ providedIn: 'root' })
export class AccountService {
  private baseUrl = `${environment.apiBaseUrl}/accounts`;

  constructor(private http: HttpClient) {}

  getAllAccounts(): Observable<Account[]> {
    return this.http.get<Account[]>(this.baseUrl);
  }

  getAccountByNumber(accountNumber: string): Observable<Account> {
    return this.http.get<Account>(`${this.baseUrl}/${accountNumber}`);
  }

  createAccount(request: CreateAccountRequest): Observable<Account> {
    return this.http.post<Account>(this.baseUrl, request);
  }

  updateAccount(accountNumber: string, request: UpdateAccountRequest): Observable<Account> {
    return this.http.put<Account>(`${this.baseUrl}/${accountNumber}`, request);
  }

  deleteAccount(accountNumber: string): Observable<void> {
    return this.http.delete<void>(`${this.baseUrl}/${accountNumber}`);
  }
}

File: backend/src/main/java/com/chhavi/smartbanking/service/AccountService.java
package com.chhavi.smartbanking.service;

import com.chhavi.smartbanking.dto.request.CreateAccountRequest;
import com.chhavi.smartbanking.dto.request.UpdateAccountRequest;
import com.chhavi.smartbanking.dto.response.AccountResponse;

import java.util.List;

public interface AccountService {
    AccountResponse createAccount(CreateAccountRequest request);
    List<AccountResponse> getAllAccounts();
    AccountResponse getAccountByAccountNumber(String accountNumber);
    AccountResponse updateAccount(String accountNumber, UpdateAccountRequest request);
    void deleteAccount(String accountNumber);
}